source/tests/yats-test/parallel_sort.h

```diff
@@ -0,0 +1,146 @@
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+#include <yats/Output.h>
+#include <yats/Input.h>
+
+namespace yats
+{
+	/// <summary>
+	/// <para>parallel_sort has been programmed demonstrate the use and construction
+	/// of the pipeline. parallel_sort sorts the elements in a vector with the aim
+	/// to utilize more than one thread in the future.</para>
+	/// <para>There are three tasks. split_task defines a vector which is supposed to be sorted.
+	/// This vector is split into two vectors of about the same size.</para>
+	/// <para>These two vectors are sorted individually in sort_task. This step can be parallized.</para>
+	/// <para>In join_task the two vectors are merged again and the sorted sequence is shown on the console.</para>
+	/// </summary>
+	namespace parallel_sort
+	{
+		/// <summary>
+		/// <para>Prints the elements of a vector to the console</para>
+		/// </summary>
+		/// <param name = "v">Vector to print to the console</param>
+		void print_vector(const std::vector<int>& v)
+		{
+			for (auto e : v)
+				std::cout << e << " ";
+			std::cout << std::endl;
+		}
+
+		/// <summary>
+		/// <para>Task to split a defined vector into two vectors of equal size +/- one.</para>
+		/// </summary>
+		class split_task
+		{
+		public:
+			split_task() : m_numbers(std::vector<int> { 5, 7, 6, 3, 35, 0, 1, 76, 9 }) {}
+
+			~split_task() = default;
+
+			/// <summary>
+			/// <para>Splits m_numbers into two vectors of about equal size. These two vectors
+			/// are output 0 and output 1.</para>
+			/// </summary>
+			yats::OutputBundle<yats::Output<std::vector<int>, 0>, yats::Output<std::vector<int>, 1>> run()
+			{
+				std::cout << "Running split_task" << std::endl;
+				std::cout << "Vector to sort: ";
+				print_vector(m_numbers);
+
+				std::vector<int> v1;
+				std::vector<int> v2;
+
+				size_t half_size = m_numbers.size() / 2;
+
+				std::copy_n(m_numbers.cbegin(), half_size, std::back_inserter(v1));
+				//print_vector(v1);
+
+				std::copy(m_numbers.cbegin() + half_size,
+					m_numbers.cend(), std::back_inserter(v2));
+				//print_vector(v2);
+
+				return std::make_tuple(v1, v2);
+			}
+
+		protected:
+			/// <summary>
+			/// <para>Vector to split.</para>
+			/// </summary>
+			const std::vector<int> m_numbers;
+		};
+
+		/// <summary>
+		/// <para>Task to sort the input vector.</para>
+		/// </summary>
+		class sort_task
+		{
+		public:
+			sort_task() = default;
+			~sort_task() = default;
+
+			/// <summary>
+			/// <para>Sorts the input vector.</para>
+			/// </summary>
+			/// <param name = "v">Vector to sort</param>
+			yats::OutputBundle<yats::Output<std::vector<int>, 0>> run(yats::Input<std::vector<int>, 0> v)
+			{
+				std::cout << "Running sort_task" << std::endl;
+				// FRAGE: Brauche ich die Konstruktion ((std::vector<int>&)v) ?
+				std::sort(((std::vector<int>&)v).begin(), ((std::vector<int>&)v).end());
```

Comment:
This is not a construction, it is a cast. To remove this, someone needs to implement operator * and/or operator->